src/task_run_diagnostic_reports.py

```diff
@@ -35,10 +37,18 @@ def success_callback(report, version):
                                          config,
                                          *additional_recipients)
 
+        # get broker host
+        match = re.match(r".*\@([^\/]*)\/*.*", config.broker)
```

Comment:
This is clumsily extracting the hostname from the configured broker (the redis location) as we need to host to spin up the Redis client in the repository object. I'm assuming here that were using the default Redis post (6379) and that won't be specified in broker. To avoid this ugliness, we could configure host, port, user and protocol separately. Maybe one for another ticket?

Reply:
Agree makes sense to configure host, port, user separately Just looking at the current config which has `broker`, `backend` - these always have to be redis now right, otherwise this will not work? And we're always using "guest" as the user, and the redis db is always called "0"? So maybe for now all we need is `host: localhost` or `host: montagu_mq_1` in the config?

Seems like a sufficiently small edit that it would be worth making here before merging, if you can stand to make any further changes ;)